cvm/quantization/tfm_ops.py

```diff
@@ -173,6 +207,53 @@ def validate(self, op, **kwargs):
 
         return op
 
+    def rewrite(self, op, **kwargs):
+        params, graph = kwargs['params'], kwargs['graph']
+        infer_shapes = kwargs['infer_shapes']
+        name, attr = op.attr('name'), op.list_attr()
+        childs = sym_iter(op.get_children())
+        pool_type = get_attr(attr, 'pool_type', 'max')
+        is_global = get_attr(attr, 'global_pool', False)
+        if pool_type == 'avg' and is_global:
+            X = childs[0]
+            X_name = X.attr('name')
+            X_shape = infer_shapes[X_name][get_entry_id(X)]
+            scale_name = X_name + '_avg_scale'
+            assert scale_name not in graph
+            graph[scale_name] = scale_sym = mx.sym.var(scale_name, shape=(1,))
+            params[scale_name] = nd.array([1. / (X_shape[2] * X_shape[3])])
+            op = mx.sym.sum(childs[0], axis=(2, 3))
+            op = mx.sym.broadcast_mul(op, scale_sym)
+        elif pool_type == 'avg':
+            X = childs[0]
+            X_shape = infer_shapes[X.attr('name')][get_entry_id(X)]
+            in_channel = X_shape[1]
+            kernel = get_attr(attr, 'kernel')
+            if isinstance(kernel, int):
+                kernel = (kernel, kernel)
+            conv_attr = {
+                'no_bias': 'True',
+                'dilate': '(1, 1)',
+                'kernel': kernel,
+                'stride': attr['stride'],
+                'pad': attr['pad'],
+                'layout': 'NCHW',
+                'num_filter': in_channel,
+                'num_group': in_channel,
+            }
+            conv_name = name.replace('pool', 'pool_conv')
+            W_name = conv_name + '_weight'
+            assert W_name not in graph
+            W_shape = (in_channel, 1, *kernel)
+            graph[W_name] = W = mx.sym.var(W_name, shape=W_shape)
+            params[W_name] = nd.full(shape=W_shape, val=(1/np.product(kernel)))
+            op = mx.sym.Convolution(X, W, **conv_attr, name=conv_name)
+        else:
+            assert pool_type == 'max', "Unsupported Pooling \
+                    %s(%s, pool_type=%s)"%(Pooling.op_name, name, pool_type)
```

Comment:
The pool_type has been validated in `validate` pass.

Reply:
updated